src/core/WorkflowContext.ts

```diff
@@ -365,7 +370,7 @@ export class WorkFlowContext {
             configFile = ensureUserPath(this.tsConfig);
         } else {
             url = path.join(this.homeDir, "tsconfig.json");
-            let tsconfig = findFileBackwards(url, appRoot.path);
```

Comment:
it uses the single app root, which is specified in 1 place (context) and can be changed by client when needed.